functions/src/validations/campaignInput.js

```diff
@@ -0,0 +1,37 @@
+const _ = require('lodash');
+const { input } = require('../config/constant');
+const { commercialId,duration, locationsSelected, status, title, validity } = input;
+/**
+ * An helper function to sort and compare data in an array
+ * @function
+ * @param [array] expected data - expected data array
+ * @param [array] input data - input data array
+ * @return  boolean result
+ */
+const arrayCompare = (expectedData, inputData) => {
+    const stringExpectedData = expectedData.sort();
+    const stringInputData = inputData.sort();
+    return JSON.stringify(stringExpectedData) === JSON.stringify(stringInputData);
+};
+/**
+ * An helper function to validate campaign input
+ * @function
+ * @param [array] datas - data to validata
+ * @return  {object} result
+ */
+const validateCampaignInput = datas => {
+    const errors = {};
+    const expectedData = [ commercialId, duration, locationsSelected, status, title, validity ];
+    const key = arrayCompare(expectedData, Object.keys(datas));
```

Comment:
can you explain what is happening from line 24 - 25?

Reply:
This just taking an array of expected keys for the campaign input and check if what is inputted campaign data is as expected